src/routes/api/db-migrations/db-migration-utils.ts

```diff
@@ -0,0 +1,90 @@
+import { Request } from "express";
+import fs from "fs";
+import path from "path";
+import { isNodeDev, isNodeTest, isNodeProd, getNodeEnv } from "utils/is-node-env";
+import { getLogger } from "config/logger";
+import { exec as execOrigin } from "child_process";
+import { promisify } from "util";
+
+const exec = promisify(execOrigin);
+const logger = getLogger("DBMigration");
+
+/**
+ * Return the common param in req.body `targetScript` to migrate up or rollback
+ * Note: Intentially to make this param case sensitive to be more safe
+ */
+export const getTargetScript = (req: Request) => {
+	let targetScript = (req.body || {}).targetScript;
+	if (!targetScript) {
+		throw {
+			statusCode: 400,
+			message: `"targetScript" is mandatory in the request body, but found none.`
+		};
+	}
+	//just make sure the script name matching the ones in db down the track.
+	if (!targetScript.endsWith(".js")) targetScript = targetScript + ".js";
+	return targetScript;
+};
+
+/**
+ * Make sure the script to migrate up or rollback is the lastest scripts in this build.
+ * So following scenarios will failed.
+ * Person A merge script 1, person B merge scripts 2. Now we CANNOT migrate anymore. Need to revert one PR first inorder to migrate.
+ *
+ * In short, encourage ppl to over-communicate when it comes to db migrations.
+ */
+export const validateScriptLocally = async (targetScript: string) => {
+
+	const scripts = await fs.promises.readdir(path.resolve(process.cwd(), "db/migrations"));
+	//Sort by name, asc, so filename order has to be in order now.
+	//So this now will be mandatory that, db migrtions scripts has to be alphabetically ordered.
```

Comment:
`db migrtions scripts has to be alphabetically ordered` <-- I prefer this and think this requirement should be fine. But thoughts welcome .

Reply:
Can it be expressed as a unit-test instead of a comment?

Reply:
Let me think about the test. But on the other hand, this is more like a "got-cha" that developer needs to remember the scripts need to be named in alphabetical order. 

I can, however, remove the restriction on this one, but that means you may "accidentally" run other's db migration scripts as well. A bit hard to explain in pure text, though.

Reply:
As long as we have a convention where the filename starts with the current date, for instance `2022-09-20`, we should be ok.

Reply:
@gxueatlassian I just don't like comments in code ;) If this is something important, I think this should be covered by a test in case someone modifies it the test will break.

Order in this case means "latest". If you think it is not clear from the top comment to the function itself, I'd've added more explanations there; at least the documentation would be in one place, more cohesive IMHO

Reply:
@bgvozdev sounds good, let's chat tomorrow to pair on this comment.